### ISSUE TITLE
New Rule: Groq API Key (@secretlint/secretlint-rule-groq)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1446]
## Description

Add a new rule to detect Groq API keys.

Groq API keys use the `gsk_` prefix followed by exactly 52 alphanumeric characters, making them reliably detectable with low false positive rate.

## Source

- [Groq Quickstart documentation](https://console.groq.com/docs/quickstart) — **does not officially document the `gsk_` prefix or token format**
- [GitHub Secret Scanning Push Protection: supported](https://docs.github.com/en/code-security/secret-scanning/introduction/supported-secret-scanning-patterns)
- [TruffleHog groq detector](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/groq/groq.go) — community-observed prefix, regex: `gsk_[a-zA-Z0-9]{52}`

## Format

Prefix `gsk_` followed by 52 alphanumeric characters (based on community observation).

Regex (with Unicode boundary and trailing boundary, ReDoS-safe):
```
/(?<!\p{L})gsk_[a-zA-Z0-9]{52}(?![a-zA-Z0-9])/gu
```

## Analysis

- The `gsk_` prefix does not collide with any other known service (IBM GSKit uses it only as a function-name component, not as a token prefix).
- A fixed 52-character body plus a unique prefix keeps FP risk low.
- Character set is limited to `[a-zA-Z0-9]` (no `+/=`).
- Added trailing boundary `(?![a-zA-Z0-9])` so the reported range ends exactly at the 52nd character.

## OK (should NOT be detected)

```
# placeholder / too short
gsk_abc123
```

## NG (should be detected)

```
GROQ_API_KEY=gsk_aBcDeFgHiJkLmNoPqRsT1234567890abcdefghijklmnopqrstuv
```
